pkg/channel/event_receiver.go

```diff
@@ -251,19 +252,22 @@ func (r *EventReceiver) ServeHTTP(response nethttp.ResponseWriter, request *neth
 		response.WriteHeader(nethttp.StatusBadRequest)
 		return
 	}
-
+	imc, err := r.imc.Get(channel.Name)
+	if err != nil {
+		r.logger.Warn("Failed to retrieve name", zap.Error(err))
+		response.WriteHeader(nethttp.StatusBadRequest)
+		return
+	}
```

Comment:
We should keep the event_receiver independent from the channel type, so this can be reused by other channel implementations.

I don't have anything specific in mind yet, but you could do the following to get the policies, which apply for the underlying channel:
* Have the underlying channels GroupVersion (e.g. set via an `EventReceiverOption()` func, when an EventReceiver is created via `NewEventReceiver()`)
* Create a Channelable lister for the GroupVersionResource (via `channelable.Get(ctx).`)
* Use this lister to load the Channelable which has the applying policies in `.status.policies`

@pierDipi do you see a more direct approach for getting the applied policies of the underlying channel?
*

Reply:
I think another approach to completely decouple and remove the `channelable` complexity too, would be:

```golang

// EventReceiver starts a server to receive new events for the channel dispatcher. The new
// event is emitted via the receiver function.
type EventReceiver struct {
	// ...

	getPoliciesFor                  func(name string) ([]duckv1.AppliedEventPolicyRef, err)
}

// ...
// usage
	policies, err := r.getPoliciesFor(channel.Name)
	if err != nil {
		r.logger.Warn("Failed to retrieve name", zap.Error(err))
		response.WriteHeader(nethttp.InternalServerError)
		return
	}

err = r.tokenVerifier.VerifyRequest(ctx, features, &r.audience, channel.Namespace, policies, request, response)
//
```

Reply:
Also because the component using the EventReceiver has most likely already its own resource lister, so it's also more efficient

Reply:
I like this approach! My only issue with this initially was, that the other implementations (rabbitMQ, NATSS ?!?) need then to implement/register this `getPoliciesFor` method/callback. 
But we could also do the AuthZ check only in case the `getPoliciesFor` callback is set, otherwise we do only AuthN :shrug:

Reply:
in case it's not set, I'd fail fast so that's it's clear that is a **must** ?

Reply:
Ah ok. As it's behind the feature flag check (`if features.IsOIDCAuthentication() {`), we should be fine if we "break" other implementations

Reply:
@7h3-3mp7y-m4n does this help and make it clearer?

Reply:
Yes @creydr and @pierDipi